ch_tools/monrun_checks/ch_backup.py

```diff
@@ -79,21 +79,37 @@ def backup_command(
     """
     Check ClickHouse backups: its state, age and count.
     """
-    backups, result = _get_backups()
+    try:
+        backups = get_backups()
+        result = Result(OK)
+    except Exception:
+        backups = []
+        result = Result(CRIT, "Failed to get backups")
+        logging.exception(result.message)
 
-    if result.code == OK:
-        result = _merge_results(
-            _check_valid_backups_exist(backups),
-            _check_last_backup_not_failed(backups, failed_backup_count_crit_threshold),
-            _check_backup_age(
-                backups, backup_age_warn_threshold, backup_age_crit_threshold
-            ),
-            _check_backup_count(backups, backup_count_warn_threshold),
-            _check_restored_data(),
-        )
+    try:
+        if result.code == OK:
+            result = _merge_results(
+                _check_valid_backups_exist(backups),
+                _check_last_backup_not_failed(
+                    backups, failed_backup_count_crit_threshold
+                ),
+                _check_backup_age(
+                    backups, backup_age_warn_threshold, backup_age_crit_threshold
+                ),
+                _check_backup_count(backups, backup_count_warn_threshold),
+                _check_restored_data(),
+            )
+    except Exception:
+        result = Result(CRIT, "Failed to check backups")
+        logging.exception(result.message)
```

Comment:
Previously, `ch-backup list` failure can result to unknown error:

```
# ch-monitoring backup
1;Unknown error: 'NoneType' object is not iterable
```

With the fix:
```
# ch-monitoring backup
2;Failed to get backups
```